src/xmipp/libraries/reconstruction/resolution_fso.cpp

```diff
@@ -25,1334 +25,1078 @@
+		MultidimArray<double> &	aniParam, double thrs)
+{
+	double N = 0;
+	for (size_t k = 0; k<aniParam.nzyxdim; k++)
+	{
+		if (DIRECT_MULTIDIM_ELEM(FSC, k) >= thrs)
 		{
-			if (DIRECT_MULTIDIM_ELEM(FSC, k) >= thrs)
-			{
-				DIRECT_MULTIDIM_ELEM(aniParam, k) += 1.0;
-				N++;
-			}
+			DIRECT_MULTIDIM_ELEM(aniParam, k) += 1.0;
+			N++;
 		}
-		DIRECT_MULTIDIM_ELEM(directionAnisotropy, dirnumber) = N;
-    }
+	}
+	DIRECT_MULTIDIM_ELEM(directionAnisotropy, dirnumber) = N;
+}
 
-    void anistropyParameterSimple(const MultidimArray<double> FSC,
-			MultidimArray<double> &aniParam, double thrs)
-    {
 
-		for (size_t k = 0; k<aniParam.nzyxdim; k++)
-			if (DIRECT_MULTIDIM_ELEM(FSC, k) >= thrs)
-				DIRECT_MULTIDIM_ELEM(aniParam, k) += 1.0;
-    }
+void ProgFSO::prepareData(MultidimArray<double> &half1, MultidimArray<double> &half2)
+{
+	Image<double> mask;
+	MultidimArray<double> &pmask = mask();
 
-    void prepareData(FileName &fnhalf1, FileName &fnhalf2,
-    		MultidimArray<double> &half1, MultidimArray<double> &half2, bool test)
-    {
-    	MultidimArray<double> &phalf1 = half1, &phalf2 = half2;
+	
+	std::cout << "Reading data..." << std::endl;
+	Image<double> imgHalf1, imgHalf2;
+	imgHalf1.read(fnhalf1);
+	imgHalf2.read(fnhalf2);
 
-    	Image<double> mask;
-    	MultidimArray<double> &pmask = mask();
+	half1 = imgHalf1();
+	half2 = imgHalf2();
 
-    	if (test)
-		{
-			Monogenic mono;
-			std::cout << "Preparing test data ..." << std::endl;
-			size_t xdim = 256, ydim = 256, zdim = 256;
-			double wavelength = 5.0, mean = 0.0, std = 0.5;
-			int maskrad = 125;
-			half1 = mono.createDataTest(xdim, ydim, zdim, wavelength, mean, 0.0);
-			half2 = half1;
-
-			mono.addNoise(phalf1, 0, std);
-			mono.addNoise(phalf2, 0, std);
-			FileName fn;
-			Image<double> saveImg;
-			fn = formatString("inputVol1_large.vol");
-			saveImg() = half1;
-			saveImg.write(fn);
-			fn = formatString("inputVol2_large.vol");
-			saveImg() = half2;
-			saveImg.write(fn);
-		}
-		else
+	// Applying the mask
+	if (fnmask!="")
+	{
+		mask.read(fnmask);
+		FOR_ALL_DIRECT_ELEMENTS_IN_MULTIDIMARRAY(pmask)
 		{
-			std::cout << "Reading data..." << std::endl;
-			Image<double> imgHalf1, imgHalf2;
-			imgHalf1.read(fnhalf1);
-			imgHalf2.read(fnhalf2);
-
-			half1 = imgHalf1();
-			half2 = imgHalf2();
-
-			if (! fnmask.isEmpty())
-			{
-				mask.read(fnmask);
-				FOR_ALL_DIRECT_ELEMENTS_IN_MULTIDIMARRAY(pmask)
-				{
-					double valmask = (double) DIRECT_MULTIDIM_ELEM(pmask, n);
-					DIRECT_MULTIDIM_ELEM(phalf1, n) = DIRECT_MULTIDIM_ELEM(phalf1, n) * valmask;
-					DIRECT_MULTIDIM_ELEM(phalf2, n) = DIRECT_MULTIDIM_ELEM(phalf2, n) * valmask;
-				}
-			}
-			mask.clear();
-			pmask.clear();
+			double valmask = (double) DIRECT_MULTIDIM_ELEM(pmask, n);
+			DIRECT_MULTIDIM_ELEM(half1, n) = DIRECT_MULTIDIM_ELEM(half1, n) * valmask;
+			DIRECT_MULTIDIM_ELEM(half2, n) = DIRECT_MULTIDIM_ELEM(half2, n) * valmask;
 		}
+	}
+	mask.clear();
```

Comment:
no need, these will be destroyed at the end of the method

Reply:
Check now, because due to that, I put the Image<double> mask inside the in condition above.